datasets/noaa/pipelines/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,1043 @@
+        df = reorder_headers(df, reorder_headers_list=reorder_headers_list)
+    if pipeline_name == "GSOD stations":
+        df = slice_column(df, slice_column_list)
+        df = filter_null_rows(df, null_rows_list=null_rows_list)
+        df = add_metadata_cols(df, source_url=source_url)
+        df = reorder_headers(df, reorder_headers_list=reorder_headers_list)
+        df["lat"] = df["lat"].astype(str)
+        df["lon"] = df["lon"].astype(str)
+        df = apply_regex(df, regex_list)
+    if pipeline_name == "GHCND hurricanes":
+        df.columns = df.columns.str.lower()
+        df = rename_headers(df, rename_headers_list=rename_headers_list)
+        df = add_metadata_cols(df, source_url=source_url)
+        df = reorder_headers(df, reorder_headers_list=reorder_headers_list)
+    if pipeline_name == "NOAA lightning strikes by year":
+        df.columns = df.columns.str.lower()
+        df = rename_headers(df, rename_headers_list=rename_headers_list)
+        df = convert_date_from_int(df, int_date_list=int_date_list)
+        df = generate_location(df, gen_location_list=gen_location_list)
+        df = add_metadata_cols(df, source_url=source_url)
+        df = reorder_headers(df, reorder_headers_list=reorder_headers_list)
+    save_to_new_file(df, file_path=str(target_file_batch))
+    append_batch_file(target_file_batch, target_file, skip_header, not (skip_header))
+
+
+def convert_date_from_int(df: pd.DataFrame, int_date_list: dict) -> pd.DataFrame:
+    logging.info("Converting dates from integers")
+    for key, values in int_date_list.items():
+        dt_col = key
+        dt_int_col = values
+        df[dt_col] = (
+            pd.to_datetime(
+                (df[dt_int_col][:].astype("string") + "000000"), "raise", False, True
+            ).astype("string")
+            + " 00:00:00"
+        )
+    return df
+
+
+def generate_location(df: pd.DataFrame, gen_location_list: dict) -> pd.DataFrame:
+    logging.info("Generating location data")
+    for key, values in gen_location_list.items():
+        loc_col = key
+        long_col = values[0]
+        lat_col = values[1]
+        df[loc_col] = (
+            "POINT("
+            + df[long_col][:].astype("string")
+            + " "
+            + df[lat_col][:].astype("string")
+            + ")"
+        )
+    return df
+
+
+def url_directory_list(
+    source_url_path: str, file_pattern: str = ""
+) -> typing.List[str]:
+    rtn_list = []
+    url = source_url_path.replace(" ", "%20")
+    req = Request(url)
+    a = urlopen(req).read()
+    soup = BeautifulSoup(a, "html.parser")
+    x = soup.find_all("a")
+    for i in x:
+        file_name = i.extract().get_text()
+        url_new = url + file_name
+        url_new = url_new.replace(" ", "%20")
+        if file_pattern == "":
+            rtn_list.append(url_new)
+        else:
+            if re.search("" + file_pattern, file_name):
+                rtn_list.append(url_new)
+            else:
```

Comment:
unnecessary else statement?

Reply:
As-per google standards you must not have an if statement without an else.